docs/user/modelling.md

```diff
@@ -0,0 +1,128 @@
+
+```python
+from typing_extensions import Annotated  # Python<3.9 - you need to install this module
+# from typing import Annotated  # Python>=3.9
+from nautobot.circuits.models import Provider
+from nautobot_ssot.contrib import NautobotModel, CustomFieldAnnotation
+class ProviderModel(NautobotModel):
+    _model = Provider
+    _modelname = "provider"
+    _identifiers = ("name",)
+    _attributes = ("custom_field",)
+
+    name: str
+    test_custom_field: Annotated[str, CustomFieldAnnotation(name="Test")]
+```
+
+!!! note
+    This is necessary because custom field names may include spaces, which are un-representable in Python object field names unless you resort to `setattr` shenanigans, which would get us into trouble here.
+
+## Many-to-one Relationships
+
+For many-to-one relationships (i.e. [foreign keys](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_one/) or [generic foreign keys](https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/#generic-relations)) a slightly different approach is employed. We need to add a field on our model for each field that we need in order to uniquely identify our related object behind the many-to-one relationship. We can do this using a [familiar synta](https://docs.djangoproject.com/en/3.2/topics/db/queries/#lookups-that-span-relationships) of double underscore separated paths. Assuming we want to synchronize prefixes and associate them with locations, we may be faced with the problems that locations aren't uniquely identified by name alone, but rather need the location type as well, we can address this as follows.
+
+```python
+from nautobot.ipam.models import Prefix
+from nautobot_ssot.contrib import NautobotModel
+class PrefixModel(NautobotModel):
+    _model = Prefix
+    _identifiers = ("network", "prefix_length")
+    _attributes = ("location__name", "location__location_type__name")
+
+    network: str
+    prefix_length: int
+
+    location__name: str
+    location__location_type__name: str
+```
+
+Now, on model `create` or `update`, the SSoT framework will dynamically pull in the location with the specified name and location type name, uniquely identifying the location and populating the foreign key.
+
+### Special Case: Generic Foreign Key
+
+In the case of a [generic foreign key](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/), we are faced with a problem. Normally, the [content type](https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/) of a relationship field can be inferred from the model class. In the case of generic foreign keys however, this is not the case. In the example of the [IP address](https://docs.nautobot.com/projects/core/en/stable/models/ipam/ipaddress/?h=ip+address) model in Nautobot, the `assigned_object` field can point to either a device's or a VM's interface. We address this the following way:
```

Comment:
```suggestion
In the case of a [generic foreign key](https://docs.djangoproject.com/en/3.2/topics/db/examples/many_to_many/), we are faced with a problem. Normally, the [content type](https://docs.djangoproject.com/en/3.2/ref/contrib/contenttypes/) of a relationship field can be inferred from the model class. In the case of generic foreign key**s** however, this is not the case. In the example of the [IP address](https://docs.nautobot.com/projects/core/en/stable/models/ipam/ipaddress/?h=ip+address) model in Nautobot, the `assigned_object` field can point to either a device's or a VM's interface. We address this the following way:
```

Reply:
I don't understand, why is it important to highlight the "s" here?

Reply:
First you write:

> In the case of a [generic foreign key]

Then you write:

> In the case of generic foreign keys however

After reading that 5 times I am still not sure what two things are being compared. It seemed logical that you were comparing `generic foreign key` to `generic foreign keys` based on the paragraph structure ... but perhaps that is wrong?? So emphasis is my best guess at what is being stated as different.

Reply:
I would say a common theme is to be explicit, often by removing pronouns but in this case a bit different. It's very complex topic and I am not always understanding "what exactly" is being referred to.